lib/rules/no-restricted-properties.js

```diff
@@ -51,36 +68,79 @@ module.exports = {
-                restrictions.set(objectName, new Map());
+            if (typeof objectName === "undefined") {
+                globallyRestrictedProperties.set(propertyName, {message: option.message});
+            } else if (typeof propertyName === "undefined") {
+                globallyRestrictedObjects.set(objectName, {message: option.message});
+            } else {
+                if (!restrictedProperties.has(objectName)) {
+                    restrictedProperties.set(objectName, new Map());
+                }
+
+                restrictedProperties.get(objectName).set(propertyName, {
+                    message: option.message
+                });
             }
+        });
+
+        /**
+        * Checks to see whether a property access is restricted, and reports it if so.
+        * @param {ASTNode} node The node to report
+        * @param {string} objectName The name of the object
+        * @param {string} propertyName The name of the property
+        * @returns {undefined}
+        */
+        function checkPropertyAccess(node, objectName, propertyName) {
+            const matchedObject = restrictedProperties.get(objectName);
+            const matchedObjectProperty = matchedObject ? matchedObject.get(propertyName) : globallyRestrictedObjects.get(objectName);
+            const globalMatchedProperty = globallyRestrictedProperties.get(propertyName);
 
-            restrictions.get(objectName).set(propertyName, {
-                message: option.message
-            });
+            if (matchedObjectProperty) {
+                const message = matchedObjectProperty.message ? " " + matchedObjectProperty.message : "";
 
-            return restrictions;
-        }, new Map());
+                context.report(node, "'{{objectName}}.{{propertyName}}' is restricted from being used.{{message}}", {
+                    objectName,
+                    propertyName,
+                    message
+                });
+            } else if (globalMatchedProperty) {
+                const message = globalMatchedProperty.message ? " " + globalMatchedProperty.message : "";
+
+                context.report(node, "'{{propertyName}}' is restricted from being used.{{message}}", {
+                    propertyName,
+                    message
+                });
+            }
+        }
 
         return {
             MemberExpression(node) {
                 const objectName = node.object && node.object.name;
                 const propertyName = astUtils.getStaticPropertyName(node);
-                const matchedObject = restrictedProperties.get(objectName);
-                const matchedObjectProperty = matchedObject && matchedObject.get(propertyName);
 
-                if (matchedObjectProperty) {
-                    const message = matchedObjectProperty.message ? " " + matchedObjectProperty.message : "";
+                checkPropertyAccess(node, objectName, propertyName);
+            },
+            VariableDeclarator(node) {
```

Comment:
I updated the PR to check for an `AssignmentExpression`, but what is an example of an `AssignmentPattern`?